### ISSUE TITLE
ResultSetMetadata problem with 0.8.6

### ISSUE THREAD

**Issue by butlnor** [impossibl/pgjdbc-ng#530]
This code is catastrophically slow with 0.8.6, but was fast with 0.7.1 and older:

```
ResultSet rs = ....
while (rs.next()) {
	ResultSetMetaData rsMeta = rs.getMetaData();
	for (int i = 1; i <= rsMeta.getColumnCount(); i++) {
		if (rsMeta.getColumnName(i).equals("someFieldName")) {
			return rs.getObject(i));
		}
	}
}
```

The problem is in getColumnName, which does an additional query each time for the metadata attributes, although this should already be cached, because I am iterating through the same result set. And because of unknown reasons, this metadata query is impossibly slower than the original data query, which produced the result set.

**Comment by kdubb** [comment]
@butlnor Each call to `ResultSet.getMetaData()` returns a new `ResultSetMetaData` that caches lookups.  So simply pulling the `rs.getMetaData()` out of the loop solves your problem.

**Comment by kdubb** [comment]
For example:
```java
ResultSet rs = ....
ResultSetMetaData rsMeta = rs.getMetaData();
while (rs.next()) {
	for (int i = 1; i <= rsMeta.getColumnCount(); i++) {
		if (rsMeta.getColumnName(i).equals("someFieldName")) {
			return rs.getObject(i));
		}
	}
}
```

**Comment by kdubb** [comment]
I think we can safely cache the result set for successive calls in most scenarios (certainly the one you've presented) and I'm going to do that so it is more compatible with previous versions.

**Comment by butlnor** [comment]
Thanks for taking care of this issue. Just a disclaimer, why I called getMetaData() in every iteration: because the code in our project is much more complex and convoluted and broken into many methods. In this ticket, I just wrote the minimal working code, to demonstrate the same effect. The reason is that we call some factory methods, that accept the ResultSet, but the main iteration is done elsewhere. So to get ResultMetadata only once, and distribute it all over the many methods, would require a lot of changes in our code.

So, instead of using the getMetaData at all, I fixed our problem with just reading from that optional column, and using try/catch, to catch the exception, when the column does not exist. Not a good solution, I agree, but it works, and it's much faster than calling getMetaData in every iteration.
